http-server/src/main/java/org/triplea/server/lobby/game/listing/GameReaper.java

```diff
@@ -0,0 +1,43 @@
+package org.triplea.server.lobby.game.listing;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import java.util.Collection;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/** Removes games that have not had a keep alive. */
+@AllArgsConstructor
+@Slf4j
+class GameReaper {
+  private final Cache<String, Boolean> keepAliveCache;
+
+  GameReaper(final int duration, final TimeUnit timeUnit) {
+    this(CacheBuilder.newBuilder().expireAfterWrite(duration, timeUnit).build());
+  }
+
+  /**
+   * Returns set of game ids to be reaped. These are games that have not received a keep alive
+   * within the cut-off time.
+   */
+  Collection<GameListing.GameId> findDeadGames(final Collection<GameListing.GameId> gameIds) {
```

Comment:
Any reason for needing to have `GameListing.GameId` instead of importing it?

Reply:
It's style mostly I suppose. I've a slight preference for namespacing inner classes rather than importing them. I don't think we have a strong convention implicit or explicit around this. Going forward, if to be called out all the time, we should add adopt a code convention around this (open an issue, get consensus, update the [wiki](https://github.com/triplea-game/triplea/wiki/Code-Conventions))

My 2 cents on that, I'm opposed as I had to by hand add the import statement, there are essentially no imports where I need to do that. I've the IDE setup to import automatically when there is only one choice and I've added filter lists so that nearly all imports are done automatically. To fix imports, I merely need to open a file. I don't see much value in a convention to explicitly import inner classes, IMO it's not going to save that much time and IMO reduces some clarity by being less explicit. I generally prefer inner classes to be namespaced, and if not then made to be proper stand-alone classes. I could see reasons to do that here for `GameId`, my initial inclination was to see how it develops as it's mostly meant to be internal to GameListing and hide the fact that ApiKey is attached to a game.

Regardless, in this instance, I've only weak preferences..  Updated: 3a92622a6

Reply:
Ok, I think there are a few other places as well. Can you do a quick search on "GameListing.GameId". I tend to prefer the import as it just makes the code easier to read IMO and more concise.

Reply:
There was one more place in the same class, now fixed: 
7099bc8 

The test code uses the non-import version, it's consistent there for now and not worth fixing IMO. I somewhat regret now updating it at all, this is where I start to push back. The legibility difference is absolutely minor, meanwhile this is an import that needs to be done by hand. I don't think there should be many imports of inner classes as that kinda violates encapsulation, inner classes really should be package level and utility level things, not used as public imports (otherwise, make them proper classes). So for me, the inner class is a callout that we're dealing with a very localized entity. `GameReaper` and `GameListing` are very closely related, hence the "reaching in" of `GameReaper` into `GameListing`. Having `GameId` be stand-alone I don't think would be good as it clutters up the package, and makes it seem like it would be okay to use in many places. For the most part, the API-key should not be used as part of the identity of lobby games, and that I want hidden as much as possible.

The reason game reaper uses GameId at all is convenience of API, so we can avoid having to strip api-key for the parameter call to add it back when processing the result. There's certainly a code smell here, but I'm trying to balance that against having to do a lot of iteration work on the caller side to remove/attach API-key. In another respect, the lack of import and namespacing I think calls out the code-smell, IMO the real fix is generally to eliminate that vs using an import.